app/controllers/training_modules_users_controller.rb

```diff
@@ -29,6 +29,14 @@ def mark_exercise_complete
     render 'courses/_block', locals: { block:, course: block.course }
   end
 
+  def find_slide
+    @dashboard_url = 'https://outreachdashboard.wmflabs.org/'
+    library_id = @training_library.id
+    module_id = @training_module.id
```

Comment:
`@training_library` and `@training_module` will both be `nil` unless they get set first.